### DISCUSSION TITLE
Allow board creator to control scrolling on all clients.

### DISCUSSION THREAD

**Discussion by tekhnee** [tableaunoir/tableaunoir#95] [Ideas]
As the tableau begins to get long (4–5 boards or more), students might get lost. The number on the upper-right corner is not always visible, to begin with, especially with small window sizes. In a remote class, without visual contact and all that loss of body language, much talking and scrolling would take place just to ensure that everyone is, literally, on the same page.

Instead, the board creator could be allowed to control scrolling on all clients (a kind of "presentation mode").

Since every client has, in general, a different window size, the scroll position could use the horizontal midpoint of the owner's view as reference, ensuring that it be displayed as midpoint on all other clients.

**Comment by francoisschwarzentruber** [comment]
This is an interesting feature. But I have two comments.

1. If I remember correctly you mentioned an interesting other feature: to show where the cursors of others are. For that we could image small arrows pointing in the direction where the others are. If that feature, you do not need anymore to force the same scrolling for every clients. Each client may be free to look at other parts of the board, and thus always aware of the others are. Furthermore, you mentioned also the minimap feature, which could also show where others are.
2. If you are the only person to present, you could also simply share your screen with any tool (BBB, Jitsi, Zoom, Microsoft Teams, Discord etc.).

What do you think?

**Comment by tekhnee** [comment]
> to show where the cursors of others are. For that we could image small arrows pointing in the direction where the others are. If that feature, you do not need anymore to force the same scrolling for every clients…. Furthermore, you mentioned also the minimap feature…

Yes, I suppose these are all alternative approaches to the same problem. Judging from experience with text editors, minimap would probably be the least effective: it's tiny, bulky, and slow to make out, forcing people to squint. Come think, it's one of the first features I turn off in any new text editor!

The "radar" arrows, as you point out, have the advantage of allowing participants to scroll around independently. I appreciate this freedom because it encourages everyone's engagement with the material, and allows people to "catch up" independently if anything has slipped off their working memory during the session.

On the other hand, central control of the viewport would allow the lecturer to stay focused on the discourse, rather than asking students to "turn to board #N" or "find the object that…"

There are pros and cons… I'm afraid I don't have a strong opinion on the matter.

> If you are the only person to present, you could also simply share your screen with any tool (BBB, Jitsi, Zoom, Microsoft Teams, Discord etc.).

I work mostly interactively... but your point is well-taken otherwise.

**Comment by francoisschwarzentruber** [comment]
Dear tekhnee,
thank you for your mesage. "Radar" arrows are implemented. I feel some fear about the central control of the viewport. Even they are adjusted to the middle of the screen, we are not guaranteed that all the users will see the same thing (but a solution would be to make that the other clients' viewport contains the sharer's viewport).

OK, I will not implement a minimap then. I think it is too much. However we could imagine a button that takes all the board and makes a minimap from it. Then you see the full board and you can choose the viewport. When the viewport is chosen, the board is again normal size and you can write again. Maybe more usable that the arrow < and > for navigating in all the board.

What do you think?

**Comment by tekhnee** [comment]
@francoisschwarzentruber if I understand correctly, you are suggesting a continuous rather than stepwise control for the viewport. I would personally find that extremely useful — *with* or *without* a minimap. Do you think the minimap could be implemented without significant overhead in terms of browser or server resources? (I only guess, without being certain, that it would be more economical to produce the minimap once and transmit it periodically to all clients, rather than producing it independently on each browser.) Personally I wouldn't miss a minimap — but others might find it more important. The continuous scroll control, though, would be very liberating.

Thank you — wishing you a Happy New Year!
